R/utils.R

```diff
@@ -238,3 +238,36 @@ get_r_string <- function(s, xpath = NULL) {
   is.na(out) <- is.na(s)
   out
 }
+
+#' Convert XML node to R code within
+#'
+#' NB this is not equivalent to `xml2::xml_text(xml)` in the presence of line breaks
+#'
+#' @param xml An `xml_node`.
+#'
+#' @return A source-code equivalent of `xml` with unnecessary whitespace removed.
+#'
+#' @noRd
+get_r_code <- function(xml) {
+  # shortcut if xml has line1 and line2 attrs and they are equal
+  # if they are missing, xml_attr() returns NA, so we continue
+  if (isTRUE(xml2::xml_attr(xml, "line1") == xml2::xml_attr(xml, "line2"))) {
```

Comment:
this makes it non-vectorized, where `xml_text()` was earlier -- is that intentional?

Reply:
`capture_sprintf_warning()` is already non-vectorized, so it's fine.